alpine-infra/src/main/java/alpine/persistence/AlpineQueryManager.java

```diff
@@ -114,7 +122,13 @@ public ApiKey getApiKey(final String key) {
      */
     public ApiKey regenerateApiKey(final ApiKey apiKey) {
         return callInTransaction(() -> {
-            apiKey.setKey(ApiKeyGenerator.generate());
+            String clearKey = ApiKeyGenerator.generate();
+            String hashedKey = BCrypt.hashpw(new String(Base64.getDecoder().decode(clearKey)), BCrypt.gensalt(ROUNDS))
+                    .toCharArray().toString();
+            apiKey.setKey(hashedKey);
+            apiKey.setSuffix(ApiKeyGenerator.generate(SUFFIX_LENGTH));
+            pm.makeTransient(apiKey);
+            apiKey.setKey(clearKey + apiKey.getSuffix());
```

Comment:
Any reason for generating a new random value for the suffix? Shouldn't it just be the last N characters of the API key?